pkg/infra/iac2/templates/api_method/factory.ts

```diff
@@ -13,7 +13,11 @@ interface Args {
 function create(args: Args): aws.apigateway.Method {
     return new aws.apigateway.Method(args.Name, {
         restApi: args.RestApi.id,
+        //TMPL {{- if .Resource.Raw }}
         resourceId: args.Resource.id,
+        //TMPL {{- else}}
+        //TMPL resourceId: args.RestApi.rootResourceId,
+        //TMPL {{- end }}
```

Comment:
:-\ It's a shame we have to do this. It's the second or third time we've had to, right? I wonder if it means we should consider coming up with a mechanism to make it nicer.

For example, I could see something where we define a custom interface for "resourceId provider", or something-something. idk.

We definitely don't need to do anything about it now, but do you think it's worth thinking about? It may be worth thinking about for a bit even if we don't intend to implement it, even if only to write a technical blog.

Reply:
its just in the apigatewayv1 stuff where we have had to do this so far. I dont think its worth investing in quite yet IMO to make thsi better unless we really start running into a lot of scenarios